pkg/client/dtclient/config_client.go

```diff
@@ -414,11 +414,22 @@ func (d *DynatraceClient) getObjectIdIfAlreadyExists(ctx context.Context, api ap
 	var matchingObjectsFound = 0
 	for i := 0; i < len(values); i++ {
 		value := values[i]
-		if value.Name == objectName || escapeApiValueName(ctx, value) == objectName {
-			if matchingObjectsFound == 0 {
+		if theAPI.CheckEqualFunc != nil {
+			var pl map[string]any
+			if err = json.Unmarshal(payload, &pl); err != nil {
+				return "", err
+			}
+			if theAPI.CheckEqualFunc(value.CustomFields, pl) {
 				objectId = value.Id
```

Comment:
In this case you overwrite the ID if more than one value matches, while so far we'd use the 1st ID (line 418 before, line 428 now) if more than one object match - as we seem to continue with just a warning log, I would not introduce different behavior for APIs with a check equal func and also only take the ID of the first matching object.